bigquery_etl/cli/backfill.py

```diff
@@ -425,71 +433,60 @@ def complete(ctx, qualified_table_name, sql_dir, project_id):
         sys.exit(1)
     client = bigquery.Client(project=project_id)
 
-    entries = get_entries_from_qualified_table_name(
-        sql_dir, qualified_table_name, BackfillStatus.COMPLETE.value
+    click.echo("Backfill processing (complete) started....")
+
+    backfills_to_process_dict = get_scheduled_backfills(
+        sql_dir, project_id, qualified_table_name, status="Complete"
     )
 
-    if not entries:
-        click.echo(f"No backfill to complete for table: {qualified_table_name} ")
-        sys.exit(1)
-    elif len(entries) > 1:
+    if backfills_to_process_dict:
+        entry_to_complete = backfills_to_process_dict[qualified_table_name]
+
         click.echo(
-            f"There should not be more than one entry in backfill.yaml file with status: {BackfillStatus.VALIDATED} "
+            f"Completing backfill(s) for {qualified_table_name} with entry date {entry_to_complete.entry_date}:"
         )
-        sys.exit(1)
 
-    entry_to_complete = entries[0]
-    click.echo(
-        f"Completing backfill for {qualified_table_name} with entry date {entry_to_complete.entry_date}:"
-    )
+        backfill_staging_qualified_table_name = (
+            get_backfill_staging_qualified_table_name(
+                qualified_table_name, entry_to_complete.entry_date
+            )
+        )
 
-    backfill_staging_qualified_table_name = get_backfill_staging_qualified_table_name(
-        qualified_table_name, entry_to_complete.entry_date
-    )
+        # clone production table
+        cloned_table_full_name = get_backfill_backup_table_name(
+            qualified_table_name, entry_to_complete.entry_date
+        )
+        _copy_table(qualified_table_name, cloned_table_full_name, client, clone=True)
 
-    # do not complete backfill when staging table does not exist
-    try:
-        client.get_table(backfill_staging_qualified_table_name)
-    except NotFound:
+        # copy backfill data to production data
+        start_date = entry_to_complete.start_date
+        end_date = entry_to_complete.end_date
+        dates = [
+            start_date + timedelta(i) for i in range((end_date - start_date).days + 1)
+        ]
+
+        # replace partitions in production table that have been backfilled
+        for backfill_date in dates:
+            if backfill_date in entry_to_complete.excluded_dates:
+                click.echo(f"Skipping excluded date: {backfill_date}")
+                continue
+
+            partition = backfill_date.strftime("%Y%m%d")
+            production_table = f"{qualified_table_name}${partition}"
+            backfill_table = f"{backfill_staging_qualified_table_name}${partition}"
+            _copy_table(backfill_table, production_table, client)
```

Comment:
A todo for the (near) future, not all tables being backfilled are partitioned, even though the backfill requires multiple queries. This should be one copy operation if the table is unpartitioned. No need to fix in this PR, but we should follow up on this fairly soon

Reply:
Awesome, thanks